### DISCUSSION TITLE
about the third-party JSON Libraries

### DISCUSSION THREAD

**Discussion by wanweiqiangintel** [oceanbase/oceanbase#895] [Ideas]
OceanBase 数据库在 3.1.3 版本推出 JSON 格式数据类型，其中依赖第三方json解析库rapidjson。目前市面上存在多种json库，如simdjson，其宣传性能大幅高于rapidjson。是否考虑使用不同的json库对oceanbase进行性能评估，使用性能更加强劲的json库？

**Comment by wanghangQ** [comment]
OB只用到了rapidjson的sax解析的原位模式，开发起来很容易，性能从这个benchmark来看已经较好了：https://github.com/miloyip/nativejson-benchmark
我们考虑过simdjson，但初步看其性能与cpu架构有关。
从parser的应用场景来看，只有解析json text或者json path的时候需要使用。也就是说，parser的性能对写场景确实有影响，但单纯提高parser的性能未必能取得很大的收益。而读取场景，解析后的json path会被缓存，性能主要依赖对json存储格式的优化。

**Comment by JLY2015** [comment]
如果需要实现套用simdjson解析json，建议是在基于原ob rapidjson 模式下做 simdjson 的增强，例如通过增加配置项，在 json parse 层告诉 json parse 模块选择相应的解析套件

**Comment by longdafeng** [comment]
@wanghangQ 能不能把rapidjson 做成插件形式, 用户可以自行进行额外设置,, 默认先使用rapidjson,  目前rapidjson 是mysql使用的, 目前我们还需要保持一定的兼容性

**Reply by wanghangQ** [reply]
应该是可以的，后续Json开发中可以考虑
